tests/arp/test_neighbor_mac_noptf.py

```diff
@@ -30,22 +30,33 @@ class TestNeighborMacNoPtf:
         6: {"intfIp": "fe00::1/64", "NeighborIp": "fe00::2"},
     }	
 
-    def count_routes(self, host, prefix):
+    def count_routes(self, asichost, prefix):
         # Counts routes in ASIC_DB with a given prefix
-        num = host.shell(
-                'sonic-db-cli ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(ROUTE_TABLE_NAME, prefix),
+        num = asichost.shell(
+                '{} ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(asichost.sonic_db_cli,ROUTE_TABLE_NAME, prefix),
                 module_ignore_errors=True, verbose=True)['stdout']
         return int(num)
 
-    def _check_no_bgp_routes(self, duthost):
-        # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on testbeds
-        localv6 = self.count_routes(duthost, "fc") + self.count_routes(duthost, "fe")
-        localv4 = self.count_routes(duthost, "10.") + self.count_routes(duthost, "192.168.0.")
-        allroutes = self.count_routes(duthost, "")
-        bgproutes = allroutes - localv6 - localv4 - DEFAULT_ROUTE_NUM
+    def _check_no_bgp_routes_asic(self, asichost):
```

Comment:
how about change the name as it's returning the count rather then checking.

Reply:
fixed in latest commit